sql/src/main/java/io/crate/action/sql/query/TransportQueryShardAction.java

```diff
@@ -53,71 +55,106 @@ public TransportQueryShardAction(ClusterService clusterService,
         this.transportService = transportService;
         this.clusterService = clusterService;
         executor = threadPool.executor(executorName);
-        transportService.registerHandler(transportAction, new TransportHandler());
+        transportService.registerHandler(queryTransportAction, new QueryTransportHandler());
+        transportService.registerHandler(queryScrollTransportAction, new ScrollTransportHandler());
     }
 
-    public void execute(String node, QueryShardRequest request, ActionListener<QuerySearchResult> listener) {
-        new AsyncAction(node, request, listener).start();
+    public void executeQuery(String node, final QueryShardRequest request, final ActionListener<QuerySearchResult> listener) {
```

Comment:
could probably be refactored into a ``executeLocalOrViaTransport(Runnable, transportAction, responseHandler)`` method